var/spack/repos/builtin/packages/liggghts/package.py

```diff
@@ -89,6 +93,10 @@ def edit(self, spec, prefix):
         if '+profile' in spec:
             makefile.filter(r'^(USE_PROFILE = ).*', r'\1"ON"')
 
+        if spec.variants['libs'].value != "NONE":
+            makefile.filter(r'^(BUILD_LIBRARIES = ).*',
+                            r'\1"{0}"'.format(spec.variants['libs'].value))
```

Comment:
Looking at this, I wonder if we can always build both shared and static, and avoid adding the variant completely.

Reply:
Thanks for your suggestion.

Technically yes. But the lib is requied by CFDEM package. By adding this variant, if there is existing Liggghts, Spack will install new Liggghts instance with libs ranther than using existing one. If we make lib installation implicit, the user might never know why the CFDEM not working when the have existing Liggghts and might give up using CFDEM with Spack.

Recommendation: Reserving lib variant to make it explicit and setting up the lib installation by default.